pkg/artifacts/signable.go

```diff
@@ -38,6 +45,18 @@ type Signable interface {
 	Enabled(cfg config.Config) bool
 }
 
+type ProvenanceSchema struct {
+	// Properties is the JSON Schema properties to support key-value pairs parameter.
+	Properties map[string]v1beta1.PropertySpec
+}
```

Comment:
This struct sounds more like a general schema for object type instead of something specific to provenance. Maybe rename and/or add some comments?

Reply:
Thanks for the feedback! Are you trying to say that this is a more generic struct and probably should be named as something more generic as well? I think this struct is not used anywhere else, and for now it's only used for provenance, and I don't see a trend that this struct would be used under other circumstances. So it seems reasonable to keep the naming still under the scope of provenance.

Reply:
Synced offline with @ywluogg. My concern was that the name `ProvenanceSchema` might sound like we define a new provenance schema alternative to intoto/slsa. We can come back to this later if others have better suggestions on the name.

Reply:
+1 totally. I'm happy to grab more feedback along the way for better naming here. One good thing is that this is only used in limited scope and doesn't affect external integrations. I will make the struct not exposed